### ISSUE TITLE
Warning: count(): Parameter must be an array or an object that implements Countable

### ISSUE THREAD

**Issue by radu81** [Spuds/Elk-Simple-Ads#5]
I get some errors in elkarte log (elkarte 1.1.3, PHP 7.2)
file sources/subs/Ads.subs.php line 303

`

287: | {
-- | --
288: | global $user_info;
289: | static $group_cache;
290: |  
291: | $cache_key = $allowed . $denied;
292: |  
293: | if (!isset($group_cache[$cache_key]))
294: | {
295: | $allowed = !empty($allowed) ? explode(',', $allowed) : array();
296: | $denied = !empty($denied) ? explode(',', $denied) : array();
297: |  
298: | $is_allowed = false;
299: | if (count(array_intersect($user_info['groups'], $denied)) > 0)
300: | {
301: | $is_allowed = false;
302: | }
303: | elseif (count(array_intersect($user_info['groups'], $allowed)) > 0)
304: | {
305: | $is_allowed = true;
306: | }
307: |  
308: | $group_cache[$cache_key] = $is_allowed;
309: | }
310: |  
311: | return $group_cache[$cache_key];
312: | }

`
and also on line 299 Warning: array_intersect(): Argument #1 is not an array

**Comment by Spuds** [comment]
Thats interesting ... it would indicated $user_info["groups"] is not loaded yet.

Does this happen for all users or just guests?

**Comment by radu81** [comment]
Only for guests.

**Comment by radu81** [comment]
I'm not sure I solved, for now I don't see any errorsin elkarte log. I had 3 custom positions created in Simple Ads add-on, and display the ads on portal using a php block like this:
`template_ad_position('SP_right_2');`
Theese ads were not displayed. All I did was to open and save the custom positions created in SimpleAds and ads are now visible on portal and seems that errors in log have also disappeared.
I'll let you know if I see any other errors. Thank you
